packages/core-data/src/actions.js

```diff
@@ -333,6 +334,18 @@ export function* saveEntityRecord(
 			) {
 				data = { ...data, status: 'draft' };
 			}
+
+			// We perform an optimistic update here to clear all the edits that
+			// will be persisted so that if the server filters them, the new
+			// filtered values are always accepted.
+			persistedEntity = yield select(
+				'getEntityRecord',
+				kind,
+				name,
+				recordId
+			);
+			yield receiveEntityRecords( kind, name, { ...persistedEntity, ...data }, undefined, true );
```

Comment:
data doesn't have the "raw" keys, and persistedEntity does, which means we'll end up with a persisted entity with a separate format. Do you think we should just merge into the "raw" values instead?

Reply:
It shouldn't be a problem. Merging seems riskier.

Reply:
we just have to make sure the expectations are correct in the selectors relying on that value.

Reply:
Yeah